### DISCUSSION TITLE
Braces that span lines

### DISCUSSION THREAD

**Discussion by alerque** [sile-typesetter/sile#1740] [Q&A] [answered]
Ran across this and it made me wonder: could we even pull this off if we wanted to? What changes would we have to make to get there?

![pbs_twimg_com_media_Eb8i1caU8AA4](https://user-images.githubusercontent.com/173595/86526036-8772ac80-be97-11ea-94ba-6022736f5caa.jpg)

**Comment by rjmunro** [comment]
In Unicode, there's a bunch of symbols for making these:

```text
             ⎧ Top
For example  ⎨ Middle
             ⎩ Botttom
```

Unfortunately, the way they are rendered here, they don't join up. They are listed on this page: https://unicode-search.net/unicode-namesearch.pl?term=BRACKET

I'm  not sure if using them would be part of the solution for this.

**Comment by simoncozens** [comment]
What *are* you reading?

Anyway, I would be tempted to render this as a three-column table, compute the height, and write out an SVG graphic to go in the middle column.

**Comment by alerque** [comment]
> What _are_ you reading?

OSINT Twitter.

> Anyway, I would be tempted to render this as a three-column table, compute the height, and write out an SVG graphic to go in the middle column.

Are our tables robust enough for that yet? I remember giving up pretty quickly the last time I gave them a go. I'd be happy to know that's changed.

My initial thought was this would need some kind of grouping such as allowing vboxes to be assembled and pushed in a horizontal context. I also see we have a bunch of tangentially related issues waiting for something along _those_ lines and they aren't very well cross referenced. I've never messed around with Github's "projects" before but it seems like it might be a way to organize this sort of thing.

**Comment by ctrlcctrlv** [comment]
I managed to do this using the characters.

The biggest problem is that there's seemingly no way to center a table. You need to use a script, or eyeball it. (Remember kids: [Never spend 6 minutes doing something by hand when you can spend 6 hours failing to automate it.](https://twitter.com/zhuowei/status/1254266079532154880?lang=en))

Relevant portion:
```tex
\define[command=mathfourtd]{\td{\font[family=MathJax_Size4]{\process}}}
\script[src=simpletable]
\script{
-- From sile-project/sile documentation/macros.xml
-- I think probably us users aren't supposed to be using this yet lol
local simpletable = SILE.require("simpletable")
simpletable.init(SILE.documentState.documentClass, {
  tableTag = "table",
  trTag = "tr",
  tdTag = "td"
})
}
...
\begin{table}%
\tr{\font[size=18pt,weight=400]\mathfourtd{\glue[width=4.15cm]⎛}\td{\glue[width=2.3cm]\cjk{\rubyA[reading=BĒNG]{孟}\rubyA[reading=HOK]{福}\rubyA[reading=LÊ]{黎}} }\mathfourtd{⎞}}
\tr{\mathfourtd{\glue[width=4.15cm]\rebox[width=0pt]{⎜}\rebox[width=0pt]{\lower[height=5pt]{⎜}}\raise[height=5pt]{⎜}}\td{\font[size=18pt,weight=400]\cjk{\glue[width=0.3cm]フレッド・ブレンナン}}\mathfourtd{\rebox[width=0pt]{\lower[height=5pt]{⎟}}\raise[height=5pt]{⎟} }}
\font[family=Inconsolata]
\tr{\mathfourtd{\glue[width=4.15cm]⎝}\td{\font[size=18pt,weight=400]copypaste\font[family=Android Emoji]📧\font[family=Inconsolata]kittens.ph}\mathfourtd{⎠}}
\end{table}
\bigskip
```

![](https://user-images.githubusercontent.com/838783/87175141-36532600-c2a6-11ea-9edb-c2901c3dcb28.png)

**Comment by Omikhleia** [comment]
Do you mean...

![image](https://user-images.githubusercontent.com/18075640/158482422-aebf768f-aec0-4a19-829d-fc5e11b69ef4.png)

EDIT - Hints: No "maths". No weird "math characters". No "table" (but close to it). No "SVG" (in the stricter sense). No manual "tweaking / eyeballing" (but automation).

**Comment by alerque** [comment]
@Amikhleia This is like "show and tell" without the "show" you tease.

**Comment by Omikhleia** [comment]
No more teasing - with the modules from the above PR:
```
\parbox[valign=middle,width=20%lw]{\raggedright{\font[features=+smcp]{Civilian Central Wars}}}%
\hfill
\bracebox[bracewidth=0.8em]{\parbox[valign=middle,width=75%lw]{%
44. Spasm or Insensate War\par
43. Some Other Kinds of Controlled General War\par
42. Civilian Devastation Attack\par
41. Augmented Disarming Attack\par
40. Countervalue Salvo\par
39. Slow-Motion Countercity War\par
}}
```

**Comment by Omikhleia** [comment, marked as answer]
So my full answer to the above question would now be:

For the general use case:
- Use at least SILE 0.14.x
- Install [3rd-party package](https://github.com/Omikhleia/ptable.sile): `luarocks install --server=https://luarocks.org/dev ptable.sile`
- And then, in a document, use the code from my previous comment.

As an alternative for a table of contents (as said once earlier, kudos to @alerque for having brought such an example above to my eyes, I liked it some much that I used it in my own book made with SILE)
- Use at least SILE 0.14.x
- Install [3rd-party package](https://github.com/Omikhleia/fancytoc.sile): `luarocks install --server=https://luarocks.org/dev fancytoc.sile` (it will install the above too as a dependency)
- And then, in a document, `\fancytableofcontents` (possibly with optional parameters, see the package doc.)

Is there anything from the question that would require this issue to remain opened?
